tests/integrationv2/test_client_authentication.py

```diff
@@ -104,4 +104,53 @@ def test_client_auth_with_s2n_server_using_nonmatching_certs(managed_process, ci
     for results in server.get_results():
         assert results.exception is None
         assert results.exit_code == 255
-        assert b'Error: Mutual Auth was required, but not negotiated' in results.stderr
\ No newline at end of file
+        assert b'Error: Mutual Auth was required, but not negotiated' in results.stderr
+
+
+@pytest.mark.parametrize("cipher", CIPHERSUITES)
+@pytest.mark.parametrize("curve", CURVES)
+@pytest.mark.parametrize("tls13", TLS13)
+def test_client_auth_with_s2n_client_no_cert(managed_process, cipher, curve, tls13):
+    host = "localhost"
+    port = next(available_ports)
+
+    random_bytes = data_bytes(64)
+    client_options = ProviderOptions(
+        mode="client",
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        data_to_send=random_bytes,
+        use_client_auth=True,
+        client_key_file='../pems/rsa_1024_sha256_client_key.pem',
+        client_certificate_file='../pems/rsa_1024_sha256_client_cert.pem',
+        client_trust_store= "../pems/rsa_2048_sha256_wildcard_cert.pem",
+        insecure=False,
+        tls13=tls13)
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = "server"
+    server_options.key = "../pems/rsa_2048_sha256_wildcard_key.pem"
+    server_options.cert = "../pems/rsa_2048_sha256_wildcard_cert.pem"
+
+    # Passing the type of client and server as a parameter will
+    # allow us to use a fixture to enumerate all possibilities.
+    server = managed_process(OpenSSL, server_options, timeout=5)
+    client = managed_process(S2N, client_options, timeout=5)
+
+    # The client should connect and return without error
+    for results in client.get_results():
```

Comment:
How do we know that this was a no_cert client auth handshake?

Reply:
So i sanity checked with a print statement in testing but I need to look through the output to find the correct wording to be able to add a verification statement

Reply:
unfortunately openssl doesn't print it as clearly as s2n does :)

Reply:
Could we add the "-state" flag to openssl to be more sure we're actually testing what we intend to test here? I'm not sure "no errors" is sufficient to prove a feature works.